Update sponsors layout logic

davew2040 opened the pull request:
Two main changes here:

(1) Moves the definition of the sponsors into a Jekyll data container, which is in `_data/sponsors.yml`
(2) Changed `index.md` to iterate through these sponsor items 
(3) Added styles to render these sponsor items as a CSS grid
(4) Added some conditional media queries to show the sponsors differently if viewed on a mobile device (hopefully!)

This is what the site should look like in a desktop browser:

![image](https://github.com/user-attachments/assets/7ba3eff4-bd1a-4513-8a54-8cd6127fdf6f)

This is what it should look like on a mobile device:

![image](https://github.com/user-attachments/assets/6558dcb7-6e05-4889-85fc-af8050605a69)

andylech commented:
Thank you, @davew2040.  I like how you are pulling from the Jekyll data container and what you did with the styling.  Looks great.

So, I approve of your coding and styling, but I think I misled you in my use of terminology between sponsor and partner.  That's on me, so let me break down how I would like them organized:

Sponsors [Paid (So Far)]:
- Xebia (have logo)
- SNI (have logo)
- CyberMedics (have logo) [Paid tonight]
- DevForge (asked for logo again - still waiting)
- Code (have logo)
- DynaByte (got logo at the end of last week)

Partners [Donations (time and/or space) and Local User Groups]:
- ONETUG (have logo)
- Seminole State College (have logo)
- GDG Central Florida (have logo)
- SQLOrlando (have logo)

1.) These two groups were in two separate blocks on the bottom of the home page last year like this:
https://web.archive.org/web/20240225093138/https://orlandocodecamp.com/
[In that example, you can skip DevFest Central Florida and SQLSaturday for now as we will eventual break out events separately.]

Please adapt your changes to work with that layout on the home page with different sponsor and partner blocks like last year.

2. These two groups will stay with separate Sponsors and Partners pages on the site. (In the future, I would like to add a Local Events page with DevFestFL and SQLSaturday as mentioned above, but that can come later.)

The only recent updates to the Sponsors page needed for now are:

- The Xebia link changed from last year and is just: https://xebia.com/
- Add CyberMedics as a Silver sponsor
- Add the logo for DynaByte

Hopefully, that clarifies the changes needed. Let me know if you have any questions.

Thanks,
Andy

davew2040 commented:
> Thank you, @davew2040. I like how you are pulling from the Jekyll data container and what you did with the styling. Looks great.
> 
> So, I approve of your coding and styling, but I think I misled you in my use of terminology between sponsor and partner. That's on me, so let me break down how I would like them organized:
> 
> Sponsors [Paid (So Far)]:
> 
> * Xebia (have logo)
> * SNI (have logo)
> * CyberMedics (have logo) [Paid tonight]
> * DevForge (asked for logo again - still waiting)
> * Code (have logo)
> * DynaByte (got logo at the end of last week)
> 
> Partners [Donations (time and/or space) and Local User Groups]:
> 
> * ONETUG (have logo)
> * Seminole State College (have logo)
> * GDG Central Florida (have logo)
> * SQLOrlando (have logo)
> 
> 1.) These two groups were in two separate blocks on the bottom of the home page last year like this: https://web.archive.org/web/20240225093138/https://orlandocodecamp.com/ [In that example, you can skip DevFest Central Florida and SQLSaturday for now as we will eventual break out events separately.]
> 
> Please adapt your changes to work with that layout on the home page with different sponsor and partner blocks like last year.
> 
> 2. These two groups will stay with separate Sponsors and Partners pages on the site. (In the future, I would like to add a Local Events page with DevFestFL and SQLSaturday as mentioned above, but that can come later.)
> 
> The only recent updates to the Sponsors page needed for now are:
> 
> * The Xebia link changed from last year and is just: https://xebia.com/
> * Add CyberMedics as a Silver sponsor
> * Add the logo for DynaByte
> 
> Hopefully, that clarifies the changes needed. Let me know if you have any questions.
> 
> Thanks, Andy

Understood I think! Added `data/sponsors.yml` as a new data list for sponsors, and adjusted the main page and the sponsors page to both refer to this new data list. 

Here's the main page:

![image](https://github.com/user-attachments/assets/3a6ed2c3-f99f-4fdf-95c7-93c9c3e33570)

Here's the sponsors page:

![image](https://github.com/user-attachments/assets/9878c343-89ca-4d82-af22-a2df3c9fc6dc)

davew2040 commented:
> snip

Additionally had to include some alternate logic to show DevForge without the logo:
``` 
{% for sponsor in site.data.sponsors %}
---
## {{ sponsor.level }} Sponsor

{% if sponsor.logoPath %}
![{{ sponsor.name }}]({{ sponsor.logoPath }}){:class="logo-sponsor-page"}

[ {{sponsor.urlLabelOverride | default: "Company Site"}} ]({{ sponsor.url }})

{% else %}

[ {{sponsor.name}} ]( {{sponsor.url}} )

{% endif %}

{% endfor %}
```

Looks like this:

![image](https://github.com/user-attachments/assets/85220be0-c1ac-4867-b91f-cb4006a98d91)

andylech reviewed (approved):
Thanks, @davew2040.
I added the logo for Tim Corey's DevForge that we just got and fixed a few types.  Looks good.
